src/ui/components/post/post_listing/thumbnail.rs

```diff
@@ -29,92 +29,104 @@ enum ThumbnailWrapperType {
 struct ThumbnailData {
   pub icon: IconType,
   pub icon_size: IconSize,
-  pub wrapper_class: Rc<str>,
-  pub icon_class: Rc<str>,
+  pub wrapper_class: Arc<str>,
+  pub icon_class: Arc<str>,
 }
 
 #[component]
 pub fn Thumbnail(
-  url: Memo<Option<Rc<str>>>,
-  image_url: Memo<Option<Rc<str>>>,
+  url: Memo<Option<Arc<str>>>,
```

Comment:
Is `Arc` supposed to be used for signals / memos?

Reply:
See [the Leptos v0.7.0 release notes](https://github.com/leptos-rs/leptos/releases/tag/v0.7.0) section on Send/Sync Signals. Basically, signals need to be threadsafe now, and since this was already using the non-threadsafe `Rc`, `Arc` was the natural choice since it lets us do everything `Rc` does but threadsafe.